crates/bevy_hierarchy/src/commands.rs

```diff
@@ -0,0 +1,1167 @@
+use bevy_ecs::{
+    prelude::{Bundle, Entity, Events},
+    system::{Command, Commands, EntityCommands},
+    world::{EntityMut, World},
+};
+use smallvec::SmallVec;
+
+use crate::{Children, HierarchyEvent, Parent};
+
+/// Trait defining how to modify the hierarchy.
+pub trait HierarchyCommands {
+    /// Add the child.
+    ///
+    /// If the child already had a parent it will be removed from that parent.
+    ///
+    /// If the child already belonged to this entity it will be moved to the end of the list of children.
+    fn add_child(&mut self, child: Entity) -> &mut Self;
+
+    /// Add the children.
+    ///
+    /// If the children already had parents they will be removed from those parents.
+    ///
+    /// If the children already belonged to this entity they will be moved to the end of the list of children.
+    fn add_children(&mut self, children: &[Entity]) -> &mut Self;
+
+    /// Insert the child at the given index.
+    ///
+    /// If the child already had a parent it will be removed from that parent.
+    fn insert_child(&mut self, index: usize, child: Entity) -> &mut Self;
+
+    /// Insert the children at the given index.
+    ///
+    /// If the child already had a parent it will be removed from that parent.
+    fn insert_children(&mut self, index: usize, children: &[Entity]) -> &mut Self;
+
+    /// Remove the child.
+    fn remove_child(&mut self, child: Entity) -> &mut Self;
+
+    /// Remove the children.
```

Comment:
More complete docs here would be nicer. This seems a bit lacking.

Reply:
Sure, what would you like to see documented?